### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AndrewYongZhenNing**

### CODE REVIEW

File: physics/2pt-fit.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@ int main(int argc, char *argv[])
     // parse arguments /////////////////////////////////////////////////////////
     OptParser            opt;
     bool                 parsed, doPlot, doHeatmap, doCorr, fold;
-    string               corrFileName, model, outFileName, outFmt;
+    string               corrFileName, model, outFileName, outFmt, save;
```
Comment (aportelli) [inline physics/2pt-fit.cpp]: change this variable name to something more explicit like `savePlot`

File: physics/2pt-fit.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ int main(int argc, char *argv[])
     opt.addOption("s", "shift"    , OptParser::OptType::value  , true,
                   "time variable shift", "0");
     opt.addOption("m", "model"    , OptParser::OptType::value  , true,
-                  "fit model (exp|exp2|exp3|cosh|cosh2|cosh3|explin|<interpreter code>)", "cosh");
+                  "fit model (exp|exp2|exp3|sinh|cosh|cosh2|cosh3|coth|explin|const|<interpreter code>)", "cosh");
```
Comment (aportelli) [inline physics/2pt-fit.cpp]: remove the coth model

File: physics/2pt-fit.cpp [hunk 3]
```diff
@@ -48,6 +48,8 @@ int main(int argc, char *argv[])
                   "show the fit plot");
     opt.addOption("h", "heatmap"  , OptParser::OptType::trigger, true,
                   "show the fit correlation heatmap");
+    opt.addOption("s", "save", OptParser::OptType::value, true,
```
Comment (aportelli) [inline physics/2pt-fit.cpp]: rename the long option as `save-plot`

File: physics/2pt-fit.cpp [hunk 4]
```diff
@@ -117,7 +120,7 @@ int main(int argc, char *argv[])
     
     // make model //////////////////////////////////////////////////////////////
     DoubleModel mod;
-    bool        coshModel = false, linearModel = false;
+    bool        sinhModel = false, coshModel = false, cotanhModel = false, linearModel = false, constModel = false;
```
Comment (aportelli) [inline physics/2pt-fit.cpp]: remove `coth`

File: physics/2pt-fit.cpp [hunk 5]
```diff
@@ -174,6 +186,18 @@ int main(int argc, char *argv[])
                                  + p[5]*(exp(-p[2]*x[0])+exp(-p[4]*(nt-x[0])));
                         }, 1, nPar);
     }
+    else if (model == "coth")
```
Comment (aportelli) [inline physics/2pt-fit.cpp]: remove this model

File: physics/2pt-fit.cpp [hunk 6]
```diff
@@ -214,16 +247,40 @@ int main(int argc, char *argv[])
...
+    else if(cotanhModel)
+    {
+            // init(0) = 0.5*(data.y(0, 0)[central]+ data.y(nt-1, 0)[central]);
+            // init(1) = 0.5*(data.y(0, 0)[central]- data.y(nt-1, 0)[central]);
+        init(0) = -0.1677;
+        init(1) = 0.02526;
+        cout << init(0) << endl;
+        
+        
+     }
```
Comment (aportelli) [inline physics/2pt-fit.cpp]: remove

File: physics/2pt-fit.cpp [hunk 7]
```diff
@@ -233,28 +290,54 @@ int main(int argc, char *argv[])
...
+                // locMin.setLowLimit(p, -0.3);
+                // globMin.setHighLimit(p,1);
+                globMin.setLowLimit(p,-1);
+                locMin.setLowLimit(p, -0.3);
+                globMin.setHighLimit(p,0);
+
+                globMin.setLowLimit(p+1,-0.12);
+                locMin.setLowLimit(p+1, -0.1);
+                globMin.setHighLimit(p+1,0.9);
+        }
```
Comment (aportelli) [inline physics/2pt-fit.cpp]: remove

File: physics/eff-mass.cpp [hunk 8]
```diff
@@ -0,0 +1,116 @@
+#include <LatAnalyze/Core/OptParser.hpp>
```
Comment (aportelli) [inline physics/eff-mass.cpp]: this file is too specific to go into LatAnalyze, please move it to your analysis code

File: physics/pert-eff-mass.cpp [hunk 9]
```diff
@@ -0,0 +1,123 @@
+#include <LatCore/OptParser.hpp>
```
Comment (aportelli) [inline physics/pert-eff-mass.cpp]: this file is too specific to go into LatAnalyze, please move it to your analysis code

**Review by aportelli — CHANGES_REQUESTED** [review]
Hi Andrew, I would like to merge that, could you implement the changes in the review please?

**Comment by AndrewYongZhenNing** [comment]
Hi Antonin,

I’ve made the edits and have pushed them to the develop branch now. Let me know if there are more things to edit.

Best wishes,

Andrew Yong

**Comment by aportelli** [comment]
you did not remove the effective mass files

**Review by aportelli — APPROVED** [review]

**Review by aportelli — APPROVED** [review]

**Comment by aportelli** [comment]
All good, thanks!

**Pull request merged**